pennylane/ops/op_math/prod.py

```diff
@@ -0,0 +1,210 @@
+from functools import reduce
+
+import numpy as np
+
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix
+
+
+def prod(*ops, do_queue=True, id=None):
+    """Represent the tensor product (or matrix product) between operators."""
+    return Prod(*ops, do_queue=do_queue, id=id)
+
+
+def _prod(
+    mats_gen,
+):  # TODO: current multiplication always expands mats, this is not always necessary
+    """Multiply matrices together"""
+    res = reduce(
+        math.dot, mats_gen
+    )  # Inefficient method (should group by wires like in tensor class)
+    return res
+
+
+class Prod(Operator):
+    """Arithmetic operator subclass representing the scalar product of an
+    operator with the given scalar."""
+
+    _name = "Prod"
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *factors, do_queue=True, id=None):  # pylint: disable=super-init-not-called
+        """Initialize a Prod instance"""
+        self._id = id
+        self.queue_idx = None
+
+        if len(factors) < 2:
+            raise ValueError(f"Require at least two operators to multiply; got {len(factors)}")
+
+        self.factors = factors
+        self._wires = qml.wires.Wires.all_wires([s.wires for s in self.factors])
+
+        if do_queue:
+            self.queue()
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " @ ".join([f"{f}" for f in self.factors])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.factors = tuple(f.__copy__() for f in self.factors)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "factors"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    def terms(self):  # is this method necessary for this class?
+        return [1.0], [self]
+
+    @property
+    def data(self):
+        """Create data property"""
+        return [f.parameters for f in self.factors]
+
+    @data.setter
+    def data(self, new_data):
+        """Set the data property"""
+        for new_entry, op in zip(new_data, self.factors):
+            op.data = copy(new_entry)
+
+    @property
+    def batch_size(self):
+        """Batch size of input parameters."""
+        raise ValueError("Batch size is not defined for Prod operators.")
+
+    @property
+    def ndim_params(self):
+        """ndim_params of input parameters."""
+        raise ValueError("Dimension of parameters is not currently implemented for Prod operators.")
+
+    @property
+    def num_params(self):
+        return sum(op.num_params for op in self.factors)
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def is_hermitian(self, run_check=True):  # TODO: cache this value, this check is expensive!
```

Comment:
Cannot add arguments to a property. Maybe we could add a method `check_hermicity` that does the check and updates an `_is_hermitian` attribute (which is the one returned by the `is_hermitian` property)?

Reply:
We could pass in a `check_hermiticity` keyword at initialization time, and then cache the output of check.